plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,321 @@
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName)
+    {
+        execute(session, "CREATE DATABASE " + quoted(schemaName));
+    }
+
+    @Override
+    public void dropSchema(ConnectorSession session, String schemaName)
+    {
+        execute(session, "DROP DATABASE " + quoted(schemaName));
+    }
+
+    @Override
+    protected Optional<BiFunction<String, Long, String>> limitFunction()
+    {
+        return Optional.of((sql, limit) -> sql + " LIMIT " + limit);
+    }
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        // ClickHouse does not have schemas
+        DatabaseMetaData metadata = connection.getMetaData();
+        return metadata.getTables(
+                null,
+                schemaName.orElse(null),
+                escapeNamePattern(tableName, metadata.getSearchStringEscape()).orElse(null),
+                new String[] {"TABLE", "VIEW"});
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn,
+            String newColumnName)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            DatabaseMetaData metadata = connection.getMetaData();
+            if (metadata.storesUpperCaseIdentifiers()) {
+                newColumnName = newColumnName.toUpperCase(ENGLISH);
+            }
+            String sql = format("ALTER TABLE %s RENAME COLUMN %s TO %s ",
+                    quoted(handle.getRemoteTableName()),
+                    quoted(jdbcColumn.getColumnName()), quoted(newColumnName));
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, String catalogName, String schemaName, String tableName,
+            SchemaTableName newTable)
+    {
+        String sql = format("RENAME TABLE %s.%s TO %s.%s", schemaName, tableName, newTable.getSchemaName(),
+                newTable.getTableName());
+        execute(session, sql);
+    }
+
+    @Override
+    public boolean isLimitGuaranteed(ConnectorSession session)
+    {
+        return true;
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection,
+            JdbcTypeHandle typeHandle)
+    {
+        String jdbcTypeName = typeHandle.getJdbcTypeName()
+                .orElseThrow(() -> new TrinoException(JDBC_ERROR, "Type name is missing: " + typeHandle));
+
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+        if (jdbcTypeName.equals("UInt8")) {
+            return Optional.of(booleanColumnMapping());
```

Comment:
uInt8 doesn't sound like boolean

map to smallint

Reply:
are there other UInt types?

Reply:
boolean will be mapping to UInt8, but UInt8 may be mapping to tinyint, boolean, references [here](https://clickhouse.tech/docs/en/sql-reference/data-types/int-uint/)

Reply:
in later version, It's also supprt UInt64 and UInt256, but missing UInt128.

- UInt8 — [0 : 255]
- UInt16 — [0 : 65535]
- UInt32 — [0 : 4294967295]
- UInt64 — [0 : 18446744073709551615]
- UInt256 — [0 : 115792089237316195423570985008687907853269984665640564039457584007913129639935]

UInt128 is not supported yet.

Reply:
We don't have to support all Uint types in the first iteration, can be followup

> boolean will be mapping to UInt8, but UInt8 may be mapping to tinyint,

ok, let's store Trino boolean as unit8 value 1
and yes, it will be represented back as a number, not a boolean anymore

with different types systems, type mapping do nt roundtrip (same for eg char -> string -> varchar case)